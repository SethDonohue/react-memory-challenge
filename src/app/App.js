import React, { Component } from 'react';

import logo from '../assets/logo.svg';
import Grid from '../grid/grid';
import './App.css';

class App extends Component {
  state = {
    cardAmount: 16,
    colorSet: ['rgb(100, 0, 0)', 'rgb(0, 100, 0)', 'rgb(0, 0, 100)',
      'rgb(100, 100, 0)', 'rgb(0, 100, 100)', 'rgb(100, 0, 100)',
      'rgb(200, 0, 0)', 'rgb(0, 200, 0)', 'rgb(0, 0, 200)',
      'rgb(200, 200, 0)', 'rgb(0, 200, 200)', 'rgb(200, 0, 200)',
      'rgb(300, 0, 0)', 'rgb(0, 300, 0)', 'rgb(0, 0, 300)',
      'rgb(300, 300, 0)',
    ],
    // 'rgb(0, 300, 300)', 'rgb(300, 0, 300)',
    // 'rgb(100, 100, 100)', 'rgb(0, 0, 0)', 'rgb(200, 200, 200)',
    // 'rgb(256, 256, 256)'
  }

  // componentDidMount() {
  //   console.log('Grid Mounted');
  //   // Run createGrid() here so it only happens once, not everytme state is updated..
  //   // const grid = this.createGrid();
  //   this.setState({ //eslint-disable-line
  //     colorArr: this.randomColorArr(),
  //   });
  // }


  randomInt = (max) => Math.floor(Math.random() * Math.floor(max));

  randomColorArr = () => {
    // create 8 random colors in a map with keys
    const map = {};
    while (Object.keys(map).length < 8) {
      const randomIndex = this.randomInt(this.state.colorSet.length);
      map[(this.state.colorSet[randomIndex])] = 2;
    }
    // add arr oh keys onto self to double array length
    const mapKeys = Object.keys(map);
    const allColorsArr = mapKeys.concat(mapKeys);

    // loop through arr for length and randomly insert elements from that arr into a new tempArr
    const tempArr = [];
    while (allColorsArr.length > 0) {
      const randomIndex = this.randomInt(allColorsArr.length);
      tempArr.push(allColorsArr[randomIndex]);
      allColorsArr.splice(randomIndex, 1);
    }
    return tempArr;
  }

  render() {
    const colorArr = this.randomColorArr();

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to Memory</h1>
        </header>
        <Grid
          cardAmount={this.state.cardAmount}
          colorArr={colorArr}
        />
      </div>
    );
  }
}

export default App;
