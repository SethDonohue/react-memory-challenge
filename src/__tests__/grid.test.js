import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Grid from '../grid/grid';
import App from '../app/App';
import '../setupTests';

(function mockLocalStorage() {
  global.localStorage = { bestScore: 0 };
}());

const wrapper = shallow(<App />);
const dataArray = wrapper.instance().randomDataArray();

test('Renders a Grid and compares to snapshot', () => {
  const component = renderer.create(
    <Grid
      dataArray={dataArray}
      bestScore={0}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
