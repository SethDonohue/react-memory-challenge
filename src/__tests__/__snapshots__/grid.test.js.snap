// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a Grid and compares to snapshot 1`] = `
Array [
  <h1 />,
  <h1>
    Attempts: 
    0
     Best: 
    0
  </h1>,
  <div
    id="grid"
  >
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
    <div
      className="card"
      id={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "gray",
        }
      }
      type="button"
    >
      <p
        className="card-text"
        style={
          Object {
            "display": "none",
          }
        }
      >
        
      </p>
    </div>
  </div>,
  <button
    id="resetButton"
    onClick={[Function]}
  >
    Reset Game
  </button>,
]
`;
